extensions/control-plane/api/management-api/cel-api/src/main/java/org/eclipse/edc/virtualized/connector/controlplane/api/management/cel/v4/CelExpressionApiV4Controller.java

```diff
@@ -0,0 +1,145 @@
+import org.eclipse.edc.participantcontext.spi.types.ParticipantContext;
+import org.eclipse.edc.spi.EdcException;
+import org.eclipse.edc.spi.query.QuerySpec;
+import org.eclipse.edc.spi.result.Result;
+import org.eclipse.edc.transform.spi.TypeTransformerRegistry;
+import org.eclipse.edc.virtualized.policy.cel.model.CelExpression;
+import org.eclipse.edc.virtualized.policy.cel.service.CelPolicyExpressionService;
+import org.eclipse.edc.web.spi.exception.InvalidRequestException;
+import org.eclipse.edc.web.spi.validation.SchemaType;
+
+import static jakarta.json.stream.JsonCollectors.toJsonArray;
+import static jakarta.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.eclipse.edc.spi.query.QuerySpec.EDC_QUERY_SPEC_TYPE_TERM;
+import static org.eclipse.edc.virtualized.policy.cel.model.CelExpression.CEL_EXPRESSION_TYPE_TERM;
+import static org.eclipse.edc.web.spi.exception.ServiceResultHandler.exceptionMapper;
+
+@Consumes(APPLICATION_JSON)
+@Produces(APPLICATION_JSON)
+@Path("/v4alpha/celexpressions")
+public class CelExpressionApiV4Controller implements CelExpressionApiV4 {
+
+    private final CelPolicyExpressionService service;
+    private final TypeTransformerRegistry transformerRegistry;
+
+    public CelExpressionApiV4Controller(CelPolicyExpressionService service,
+                                        TypeTransformerRegistry transformerRegistry) {
+        this.service = service;
+        this.transformerRegistry = transformerRegistry;
+    }
+
+    @POST
+    @RolesAllowed({ParticipantPrincipal.ROLE_ADMIN})
+    @RequiredScope("management-api:write")
+    @Override
+    public JsonObject createExpressionV4(@SchemaType(CEL_EXPRESSION_TYPE_TERM) JsonObject expression) {
+
+        var expr = transformerRegistry.transform(expression, CelExpression.class)
+                .orElseThrow(InvalidRequestException::new);
+
+        service.create(expr)
+                .orElseThrow(exceptionMapper(CelExpression.class, expr.getId()));
+
+        var responseDto = IdResponse.Builder.newInstance()
+                .id(expr.getId())
+                .createdAt(expr.getCreatedAt())
+                .build();
+
+        return transformerRegistry.transform(responseDto, JsonObject.class)
+                .orElseThrow(f -> new EdcException("Error creating response body: " + f.getFailureDetail()));
+    }
+
+    @GET
+    @Path("{id}")
+    @RolesAllowed({ParticipantPrincipal.ROLE_ADMIN})
+    @RequiredScope("management-api:read")
+    @Override
+    public JsonObject getExpressionV4(@PathParam("id") String id, @Context SecurityContext securityContext) {
+
+        var expr = service.findById(id)
+                .orElseThrow(exceptionMapper(ParticipantContext.class, id));
+
+        return transformerRegistry.transform(expr, JsonObject.class)
+                .orElseThrow(f -> new EdcException("Error creating response body: " + f.getFailureDetail()));
+    }
+
+    @PUT
+    @Path("{id}")
+    @RolesAllowed({ParticipantPrincipal.ROLE_ADMIN})
```

Comment:
it might make sense to also allow the ROLE_PROVISIONER here and on other API methods, since global master data, such as CEL expressions, should be seeded by the provisioning agent